### ISSUE TITLE
residue.index does not make sense for some sliced trajectories that don't contain residue 0 

### ISSUE THREAD

**Issue by sroet** [mdtraj/mdtraj#1585]
Main issue:
```python
#Load a trajectory
full = md.load("data/4obe.pdb")
# Slice to any number of residues that don't start with index 0
selection = full.topology.select("resid 1 to 10")
sliced = full.atom_slice(selection)

 # get the index of the last residue
idx = [res.index for res in sliced.topology.residues][-1]
print(idx)
#try to get that index this fails
res = sliced.topology.residue(idx)
```
this returns
```
10

---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-3-2b59da77cad5> in <module>
      3 print(idx)
      4 #try to get that index this fails
----> 5 res = sliced.topology.residue(idx)

~/miniconda3/envs/contact_map/lib/python3.8/site-packages/mdtraj/core/topology.py in residue(self, index)
    808             The `index`-th residue in the topology.
    809         """
--> 810         return self._residues[index]
    811 
    812     @property

IndexError: list index out of range
```

I have tracked down the bug, and should be able to provide a PR with a fix soon

**Comment by sroet** [comment]
cc @dwhswenson this came up while testing out `atom_slice`d trajectories for `contact_map` (most of our residue.index assumptions break then)

**Comment by sroet** [comment]
Closed by #1586
